script/script_test.go

```diff
@@ -61,16 +62,75 @@ func TestScript(t *testing.T) {
 			context: testContext{
 				script: `function request(ctx, params); ctx.request.header["User-Agent"] = "skipper.lua/1.0"; end`,
 			},
-			expectedRequestHeader: map[string]string{"user-agent": "skipper.lua/1.0"},
+			expectedRequestHeader: http.Header{"User-Agent": []string{"skipper.lua/1.0"}},
+		},
+		{
+			name: "add request header",
+			context: testContext{
+				script: `function request(ctx, params)
+					ctx.request.header.add("Foo", "Bar")
+					ctx.request.header.add("Foo", "Baz")
+				end`,
+			},
+			expectedRequestHeader: http.Header{"Foo": []string{"Bar", "Baz"}},
+		},
+		{
+			name: "request header values",
+			context: testContext{
+				script: `function request(ctx, params)
+					ctx.request.header.add("Foo", "Bar")
+					ctx.request.header.add("Foo", "Baz")
+
+					ctx.request.header["Qux"] = table.concat(ctx.request.header.values("Foo"), ", ")
+					ctx.request.header["Absent-Length"] = table.getn(ctx.request.header.values("Absent"))
+				end`,
+			},
+			expectedRequestHeader: http.Header{
+				"Foo":           []string{"Bar", "Baz"},
+				"Qux":           []string{"Bar, Baz"},
+				"Absent-Length": []string{"0"},
```

Comment:
Should we test `Foo-Length` (I expect it to be 2) and `Qux-Length` (I expect it to be 1) here as well?

Reply:
I added this case to show that length of empty values is zero